flake.nix

```diff
@@ -1,50 +1,268 @@
 {
   description = "cardano-ledger";
 
+  inputs = {
+    haskellNix.url = "github:input-output-hk/haskell.nix";
+    haskellNix.inputs.tullia.follows = "tullia";
+    nixpkgs.follows = "haskellNix/nixpkgs-unstable";
+    iohkNix.url = "github:input-output-hk/iohk-nix";
+    flake-utils.url = "github:hamishmack/flake-utils/hkm/nested-hydraJobs";
+
+    CHaP.url = "github:input-output-hk/cardano-haskell-packages?ref=repo";
+    CHaP.flake = false;
+
+    cardano-mainnet-mirror.url = "github:input-output-hk/cardano-mainnet-mirror";
+    cardano-mainnet-mirror.flake = false;
+
+    # cicero
+    tullia.url = "github:input-output-hk/tullia";
+
+    # non-flake nix compatibility
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
+  };
+
+  outputs = inputs: let
+    supportedSystems = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-linux"
+      "aarch64-darwin"
+    ];
+  in
+    inputs.flake-utils.lib.eachSystem supportedSystems (
+      system: let
+        # setup our nixpkgs with the haskell.nix overlays, and the iohk-nix
+        # overlays...
+        nixpkgs = import inputs.nixpkgs {
+          overlays =
+            [inputs.haskellNix.overlay]
+            ++ builtins.attrValues inputs.iohkNix.overlays
```

Comment:
We will need to be careful here to import them in the right order. See https://github.com/input-output-hk/cardano-db-sync/commit/a7cd5db34782e70b0044ecb97c70654506690dc5